app/models/document.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class Document < ApplicationRecord
+  belongs_to :message
+
+  has_one_attached :attachment
+  has_many :chunks, dependent: :destroy
+
+  delegated_type :documentable, types: %w[UploadedFile]
+
+  after_commit :chunk_attachment, on: %i[create update]
+
+  CHUNK_SIZE = 2_000
+
+  private
+
+  def chunk_attachment
+    return unless attachment.attached? && chunks.empty?
+
+    content = attachment.download
+    content.each_chunk(CHUNK_SIZE).with_index do |chunk_content, chunk_index|
+      chunks.create!(chunk_index:, content: chunk_content)
+    end
+  end
+end
+
+class String
+  def each_chunk(chunk_size)
+    return enum_for(:each_chunk, chunk_size) unless block_given?
+
+    (0...length).step(chunk_size) do |start_idx|
+      yield self[start_idx, chunk_size]
+    end
+  end
```

Comment:
I have no familiarity with this repo at all, so can only comment on style stuff. This class should probably live in app/lib. Also String is quite a generic term I would veer away from tbh. Also maybe put it inside a submodule Document::Substring or something? I also cant really see where this is being used

Reply:
Yup, can probably refactor to make it better. It's used on line 21